pkg/kademlia/pslice/pslice.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pslice
+
+import (
+	"sync"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/topology"
+)
+
+// PSlice maintains a list of addresses, indexing them by their different proximity orders.
+// Currently, when peers are added or removed, their proximity order must be supplied, this is
+// in order to reduce duplicate PO calculation which is normally known and already needed in the
+// calling context.
+type PSlice struct {
+	peers []swarm.Address
+	bins  []uint
+
+	sync.Mutex
+}
+
+// New creates a new PSlice.
+func New(maxBins int) *PSlice {
```

Comment:
tbh this is a bit of a wider issue. the problem is that `swarm.Proximity` returns an `int`. and not a `uint8`. I think it would be nice to have a decision about how we treat proximity order across the whole codebase. either `Proximity` should return `uint8` or it should be `int` across the board. in any case lets have it like so for this iteration, we can do this wider change later

Reply:
ok. is there an issue for this @acud ?

Reply:
in my head yes. i'll handle it. no more issues